File: Assessment.Web/Controllers/OhmValueCalculatorController.cs
using System.Linq;
using System.Web.Mvc;
using Assessment.Domain.Ohm;
using Assessment.Domain.Resistor;
using Assessment.Domain.Resistor.Repository;
using Assessment.Web.Extensions;
using Assessment.Web.Models;

namespace Assessment.Web.Controllers
{
    public class OhmValueCalculatorController : Controller
    {
        private readonly IOhmValueCalculator _ohmValueCalculator;
        private readonly IResistorColorCodeRepository _resistorColorCodeRepository;

        // Using Poor Man's DI given the scope of the project, but would refactor to using a IoC container if/when
        // the application increased in size/complexity.
        public OhmValueCalculatorController() : 
            this(new OhmValueCalculator(new ResistorColorCodeRepository()), new ResistorColorCodeRepository())
        {
        }

        public OhmValueCalculatorController(IOhmValueCalculator ohmValueCalculator, IResistorColorCodeRepository resistorColorCodeRepository)
        {
            _ohmValueCalculator = ohmValueCalculator;
            _resistorColorCodeRepository = resistorColorCodeRepository;
        }

        [HttpGet]
        public ActionResult Index()
        {
            return View();
        }

        [HttpGet]
        public ActionResult GetDropdownValues()
        {
            var digitValues = _resistorColorCodeRepository.GetAllResistorColorCodesWithNonNullDigit().Select(x => x.ResistorColor.ToString());
            var multiplierValues = _resistorColorCodeRepository.GetAllResistorColorCodesWithNonNullMultiplier().Select(x => x.ResistorColor.ToString());
            var toleranceValues = _resistorColorCodeRepository.GetAllResistorColorCodesWithNonNullTolerance().Select(x => x.ResistorColor.ToString());

            var dropdownViewModel = new ResistorDropdownViewModel
            {
                BandAValues = digitValues,
                BandBValues = digitValues,
                BandCValues = multiplierValues,
                BandDValues = toleranceValues
            };

            return Json(dropdownViewModel, JsonRequestBehavior.AllowGet);
        }

        [HttpPost]
        public ActionResult Calculate(OhmValueCalculatorPostModel postModel)
        {
            var ohmValue = _ohmValueCalculator.CalculateOhmValues(postModel.ColorBandA, postModel.ColorBandB,
                postModel.ColorBandC, postModel.ColorBandD);

            var tolerance = _resistorColorCodeRepository
                .FindBy(ResistorColorCode.ParseResistorColor(postModel.ColorBandD))
                .Tolerance;

            return Json(new OhmValueCalculatorResponseModel
            {
                OhmValue = ohmValue.ToFormattedString(),
                Tolerance = tolerance?.ToString("P")
            });
        }
    }
}

File: Assessment.Web/scripts/OhmValueCalculatorController.js
angular.module("ohmValueCalculator", [])
.controller("ohmValueCalculatorController", ["$scope", "$http", function (scope, http) {
        scope.resistorBandADropDown = [];
        scope.resistorBandBDropDown = [];
        scope.resistorBandCDropDown = [];
        scope.resistorBandDDropDown = [];

        scope.bandASelectedItem = "";
        scope.bandBSelectedItem = "";
        scope.bandCSelectedItem = "";
        scope.bandDSelectedItem = "";

        scope.ohmValue = "";
        scope.tolerance = "";

        angular.element(document).ready(function() {
            http.get("/OhmValueCalculator/GetDropdownValues")
                .success(function (result) {
                    scope.resistorBandADropDown = result.BandAValues;
                    scope.resistorBandBDropDown = result.BandBValues;
                    scope.resistorBandCDropDown = result.BandCValues;
                    scope.resistorBandDDropDown = result.BandDValues;
                });
        });

        scope.calculate = function() {
            var payload = {
                colorBandA: scope.bandASelectedItem,
                colorBandB: scope.bandBSelectedItem,
                colorBandC: scope.bandCSelectedItem,
                colorBandD: scope.bandDSelectedItem
            };

            http.post("/OhmValueCalculator/Calculate", payload)
                .success(function(result) {
                    scope.ohmValue = result.OhmValue;
                    scope.tolerance = result.Tolerance;
                });
        };

        scope.isDropdownDisabled = function() {
            return scope.bandASelectedItem === "" || scope.bandBSelectedItem === "" || scope.bandCSelectedItem === "" || scope.bandDSelectedItem === "";
        };

        scope.isResultVisible = function() {
            return scope.ohmValue !== "" && scope.tolerance !== "";
        };
    }]);